src/backend/distributed/planner/recursive_planning.c

```diff
@@ -856,9 +879,14 @@ RecursivelyPlanDistributedJoinNode(Node *distributedNode, Query *query,
 		 *      subquery but this implies that we need to rebuild the rtable and
 		 *      re-point all the Vars to the new rtable indexes. We have not
 		 *      implemented that yet.
+		 *
+		 *      Instead we can forcefully choose recursive plan for INNER JOIN between
+		 *      recurring and nonrecurring parts
```

Comment:
Maybe we can do something simpler https://github.com/citusdata/citus/commit/6df32a5d2c7be721627ce791406d6d3c005ff2ca ?

> forcefully choose recursive plan for inner join btw recurring and non recurring parts

Also, I'd like to note this change is not only about inner joins but is about recursively planning non-recurring sub join-trees that participate in outer joins as inner rels.

Reply:
> Maybe we can do something simpler https://github.com/citusdata/citus/commit/6df32a5d2c7be721627ce791406d6d3c005ff2ca ?

I'm actually thinking that we should merge this to @onurctirtir's PR. Let's discuss on the sync soon

Reply:
We agreed on merging this into #6512, so closing this PR.